homeassistant/components/zha/__init__.py

```diff
@@ -359,6 +372,90 @@ def zdo_command(self, tsn, command_id, args):
         pass
 
 
+class ZhaEndpointEntity(entity.Entity):
+    """A base class for ZHA endpoints."""
+
+    _domain = DOMAIN
+
+    def __init__(self, endpoint, manufacturer, model, application_listener,
+                 unique_id, keepalive_interval=7200, **kwargs):
+        """Init ZHA endpoint entity."""
+        self._device_state_attributes = {}
+        ieee = endpoint.device.ieee
+        ieeetail = ''.join(['%02x' % (o, ) for o in ieee[-4:]])
+        if manufacturer and model is not None:
+            self.entity_id = "{}.{}_{}_{}_{}".format(
```

Comment:
There is no need to create such complicated entity ids. Since it has a unique ID, a unique entity ID will be assigned once and used every time. You don't have to assign one at all and it will just work.

Reply:
I wanted to keep the naming consistent with the other zha entities. Otherwise the user would have no way of identifying which entities belong together.

Reply:
For me personally, having "manufacturer" and "model" in the entity_id provides me with a better user experience correctly identifying devices, especially if you join more than one device at a time.